packages/sandbox/src/lambda_function_log_streamer.ts

```diff
@@ -0,0 +1,178 @@
+    private readonly printer: Printer
+  ) {}
+
+  setOutputLocation = (outputLocation: string) => {
+    this.logsMonitor.setOutputLocation(outputLocation);
+  };
+
+  /**
+   * Starts streaming logs in the given sandbox.
+   * @param sandboxBackendId The sandbox backend identifier.
+   * @param logsFilters Glob patterns for function names to filter.
+   */
+  startStreamingLogs = async (
+    sandboxBackendId: BackendIdentifier,
+    logsFilters?: string[]
+  ) => {
+    const backendOutput: BackendOutput =
+      await this.backendOutputClient.getOutput(sandboxBackendId);
+
+    const definedFunctionsPayload =
+      backendOutput['AWS::Amplify::Function']?.payload.definedFunctions;
+    const deployedFunctionNames = definedFunctionsPayload
+      ? (JSON.parse(definedFunctionsPayload) as string[])
+      : [];
+
+    // To use list-tags API we need to convert function name to function Arn since it only accepts ARN as input
+    const deployedFunctionNameToArnMap = await this.getFunctionArnFromNames(
+      sandboxBackendId,
+      deployedFunctionNames
+    );
+
+    if (!deployedFunctionNameToArnMap) {
+      this.printer.log(
+        `[Sandbox] Could not find any function in stack ${BackendIdentifierConversions.toStackName(
+          sandboxBackendId
+        )}. Streaming function logs will be turned off.`,
+        LogLevel.DEBUG
+      );
+      return;
+    }
+
+    for (const entry of deployedFunctionNameToArnMap) {
+      const listTagsResponse = await this.lambda.send(
+        new ListTagsCommand({
+          Resource: entry.arn,
+        })
+      );
+      const friendlyFunctionName =
+        listTagsResponse.Tags?.[TagName.FRIENDLY_NAME];
+      if (!friendlyFunctionName) {
+        this.printer.log(
+          `[Sandbox] Could not find user defined name for lambda function ${entry.name}. Logs will not be streamed for this function.`,
+          LogLevel.DEBUG
+        );
+        continue;
+      }
+
+      let shouldStreamLogs = false;
+      if (logsFilters) {
+        for (const filter of logsFilters) {
+          const pattern = new RegExp(filter);
+          if (pattern.test(friendlyFunctionName)) {
+            shouldStreamLogs = true;
+            this.printer.log(
+              `[Sandbox] Logs for function ${friendlyFunctionName} will be streamed as it matched filter '${filter}'`,
+              LogLevel.DEBUG
+            );
+            break;
+          }
+        }
+      } else {
+        // No logs filter, means we stream all logs
+        this.printer.log(
+          `[Sandbox] Logs for function ${friendlyFunctionName} will be streamed.`,
+          LogLevel.DEBUG
+        );
+        shouldStreamLogs = true;
+      }
+
+      if (shouldStreamLogs) {
+        this.logsMonitor?.addLogGroups(
+          friendlyFunctionName,
+          // a CW log group is implicitly created for each lambda function with the lambda function's name
+          `/aws/lambda/${entry.name}`
```

Comment:
_technically_ customers could override this. Not sure if it's worth another API call to lookup the log stream that's configured for the lambda?

Reply:
Correct and I think it's fine to start with this and expand later on how customers are changing their log configuration (e.g. they can also decide to base 64 encode their logs). FWIW, cdk also doesn't retrieve actual CW log group but assumes this default.